lib/index.js

```diff
@@ -583,15 +583,19 @@ function _connect(options, callback) {
     // this is usually `admin` in dev and a specific db in production
     connectOptions.authSource = config.connectOptions.authSource || config.name;
   }
+
   MongoClient.connect(options.url, connectOptions, (err, client) => {
-    if(!err && !options.init) {
-      logger.info('connecting to database: ' + options.url);
+    if(err) {
+      return callback(err);
     }
-    let db = null;
-    if(client) {
-      db = client.db(config.name);
+
+    if(!options.init) {
+      logger.info('connecting to database: ' + options.url);
     }
-    callback(err, {db, client});
+
+    const db = client.db();
```

Comment:
Ok so you did remove the client check in favor of returning an error callback early, but has this been tested with a failed auth connection to a database? I'm pretty confident it will work (possibly better), but if a connect fails there is no client, there should be an error, but I would like to know what happens if that is the case for sure. This should be pretty simple to do just enableAuth and set a bogus username and password.